hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneAddress.java

```diff
@@ -96,14 +108,37 @@ public void checkUrlTypes() throws OzoneClientException, IOException {
     Assert.assertEquals("key1/key3/key", address.getKeyName());
     Assert.assertFalse("this should not be a prefix",
         address.isPrefix());
+  }
 
+  @Test
+  public void checkPrefixUrlType() throws OzoneClientException {
     address = new OzoneAddress(prefix + "vol1/bucket/prefix");
     address.ensurePrefixAddress();
     Assert.assertEquals("vol1", address.getVolumeName());
     Assert.assertEquals("bucket", address.getBucketName());
     Assert.assertEquals("prefix", address.getKeyName());
     Assert.assertTrue("this should be a prefix",
         address.isPrefix());
+  }
+
+  @Test
+  public void checkSnapshotUrlType() throws OzoneClientException {
+    address = new OzoneAddress(prefix + "vol1/bucket/.snapshot/snap1");
+    address.ensureSnapshotAddress();
+    Assert.assertEquals("vol1", address.getVolumeName());
+    Assert.assertEquals("bucket", address.getBucketName());
+    Assert.assertEquals(".snapshot/snap1", address.getSnapshotName());
+    Assert.assertEquals(".snapshot/snap1", address.getKeyName());
+
+
+    String message = "Only a snapshot name with " +
+        "a snapshot indicator is accepted";
+
+    address = new OzoneAddress(prefix + "vol1/bucket/.snapshot");
+
+    exception.expect(OzoneClientException.class);
+    exception.expectMessage(message);
 
+    address.ensureSnapshotAddress();
```

Comment:
```suggestion
    OzoneClientException exception = assertThrows(OzoneClientException.class, () -> {
        address.ensureSnapshotAddress();
    });

    assertTrue(exception.getMessage().contains(message));
``` 

It would be nicer and simpler if you use Junit5. https://www.baeldung.com/junit-assert-exception